src/components/textarea/textarea.component.tsx

```diff
@@ -268,6 +273,11 @@ export const Textarea = React.forwardRef(
       (validationIconId && !validationOnLabel)
     );
 
+    const l = useLocale();
+    const hintString = l.hintString.primaryHintString();
+    const hintId = useRef(guid());
+    const hintDescribedBy = `${characterLimit ? hintId.current : ""}`;
```

Comment:
I've noticed that lines 276-279 are exactly the same in `textbox.component.tsx`, maybe you could move it to the `useCharacterCount` hook, and then return the `StyledCharacterCountHint` as `characterCountHint` in the same manner as `CharacterCount` is returned as `characterCount` from that hook?